homeassistant/components/webdav/camera.py

```diff
@@ -0,0 +1,242 @@
+"""
+This component models a WebDAV share full of images as a camera.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.webdav/
+"""
+
+import logging
+
+import asyncio
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.components.camera import (Camera,
+                                             PLATFORM_SCHEMA, SUPPORT_ON_OFF,
+                                             SUPPORT_STREAM)
+from homeassistant.const import (
+    CONF_HOST, CONF_USERNAME, CONF_PASSWORD, CONF_PATH, CONF_NAME, CONF_TOKEN
+)
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['webdavclient3==0.10']
+
+CONF_CERTIFICATE_PATH = "ssl_client_certificate"
+CONF_IMAGE_INTERVAL = "image_interval"
+CONF_KEY_PATH = "ssl_client_key"
+
+IMAGE_URL = '/api/camera_proxy/{0}?token={1}&frame={2}'
+STREAM_URL = '/api/camera_proxy_stream/{0}?token={1}'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_TOKEN): cv.string,
+    vol.Optional(CONF_CERTIFICATE_PATH): cv.string,
+    vol.Optional(CONF_KEY_PATH): cv.string,
+    vol.Optional(CONF_PATH, default="/"): cv.string,
+    vol.Required(CONF_IMAGE_INTERVAL): cv.time_period,
+})
+
+# This must be separate from frame interval since listing can be expensive
+SCAN_INTERVAL = timedelta(minutes=15)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a single webdav camera."""
+    from webdav3.client import Client
+    from webdav3.exceptions import WebDavException
+
+    name = config[CONF_NAME]
+    host = config[CONF_HOST]
+    path = config.get(CONF_PATH)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    token = config.get(CONF_TOKEN)
+    cert_path = config.get(CONF_CERTIFICATE_PATH)
+    key_path = config.get(CONF_KEY_PATH)
+    client = Client({
+        "webdav_hostname": host,
+        "webdav_login": username,
+        "webdav_password": password,
+        "webdav_token": token,
+        "webdav_cert_path": cert_path,
+        "webdav_key_path": key_path,
+        "webdav_root": path,
+    })
+    session = async_get_clientsession(hass)
+
+    try:
+        client.list('/')  # This will throw if we can't access the share
+        add_entities(
+            [WebDavCamera(name, client, session, config[CONF_IMAGE_INTERVAL])],
+            update_before_add=True)
+    except WebDavException as exception:
+        _LOGGER.warning(
```

Comment:
warning or error?

Reply:
[The docs](https://developers.home-assistant.io/docs/en/integration_quality_scale_index.html) specify to warn once when the device is unavailable, so I took that to mean warning.

Reply:
Yep, that's my bad. Do you need the exception logged here? Since we know already what caused it and are logging a message already.